proposal/anva-kn/README.rst

```diff
@@ -0,0 +1,112 @@
+
+However, an alternative means of installation would be git cloning the repo and installing using make.
+
+- Read/Save the data
+    - Read .txt format files
+    - Read .dat format files
+    - Read .tiff format files
+    - Read all files (.txt, .dat, tiff) inside one directory
+- Preprocess the data
+    - Remove noise
+        - FFT noise removal (TBD)
+        - Moving window smoothening function
+    - Calculate SNR
+    - Remove zeros or nans
+    - Remove cosmic rays
+- Process the data
+    - Background modeling, interpolation
+    - Peaks modeling, fitting
+    - Peak clusters modeling (TBD)
+    - PCA (TBD)
+    - NMF (TBD)
+
+In my mind, the user interface would be akin to any other python module. 
+
+.. code-block:: python
+   
+    import ramanflow as rf
+    import numpy as np
+
+    # Assume 1d array
+    example_data = rf.read('exampple.txt') # or any other format
+    preprocessed_example_data = rf.preprocess.remove_cosmic_rays(example_data) # or any other preprocessing function
+    # Data cleaning till this point should be sufficient for basic ML pipeline 
+    # Now lets say a scientist wants to know the parameters of background or peaks
+    # here he/she would create an object for background and peaks
+    background_model = rf.process.background(preprocessed_data, args=...) # will take some parameters. Ex: degree of polynomial for background interpolation
+    peaks_model = rf.process.peaks(preprocessed_data, args=...) # this one as well. Ex. peak shape function (Gaussian, Voight)
+
+
+Engineering Infrastructure
+=================
+
+
+- Build System
+    - make (test, clean)
+- Testing Framework
+    - Google Test for C++
+    - Pytest for python
+- Version Control
+    - Git
+
+Schedule
+=================
+
++-------+-------------------------------------------------------------------------+
+| Week  | To-Do                                                                   |
++-------+-------------------------------------------------------------------------+
+| 1     | Implement read module                                                   |
++-------+-------------------------------------------------------------------------+
+| 2     |   Implement Preprocessing part 1 (Noise removal)                        |
++-------+-------------------------------------------------------------------------+
+| 3     |   Implement Preprocessing part 2 (Cosmic rays removal, SNR)             |
++-------+-------------------------------------------------------------------------+
+| 4     |   Implement Processing part 1 (Background modeling and interpolation)   |
++-------+-------------------------------------------------------------------------+
+| 5     |   Implement Processing part 2 (Peak fitting)                            |
++-------+-------------------------------------------------------------------------+
+| 6     |   Binding to python, unittesting, debugging, writing basic documentation|
```

Comment:
Unit tests are usually add along with the first working functions.  The later you add unit tests, the more time it takes you to finish a project.

You do not need to write tests before features as TDD suggested, though.  In my experience TDD is usually counter-productive.